### ISSUE TITLE
Some App Crash Conditions

### ISSUE THREAD

**Issue by MatthewSloyan** [MatthewSloyan/Mobile-Applications-Development-2-Project#9]
From testing the app seems to crash if blank ingredient inputs are entered on the Search by ingredients page. 

Also when testing on Android, it seems to crash when the back button is pressed but it seems to work on Windows so it could just be a build error. I will test this.

**Comment by MatthewSloyan** [comment]
I have fixed my first problem by checking if the input is null rather than "" which has solved it.

**Comment by MatthewSloyan** [comment]
From research I have found there's a bug if a page is returned to either a tabbed or master details page it crashes on Android. A fix for this is overriding the back button and moving the task to the back of the activity stack seems to solve the problem,  which I found below at the link. It seems it's a common problem, but will need to test to see if it works.

https://forums.xamarin.com/discussion/81793/back-button-from-causes-crash-on-android-when-page-is-masterdetail

**Comment by MatthewSloyan** [comment]
From testing I found that my above thoughts were incorrect as it wasn't a bug. The cause was when a page is returned the Selected value is null which causes the crash. To fix this I have implemented a return if null on all pages which fixes the issue and now the back button can be pressed again.

![image](https://user-images.githubusercontent.com/26766108/56094005-e466f900-5ec6-11e9-8ea7-e42d7747c38a.png)

I also found another error when loading recipe information which was causing a crash. It was just to the fact that some recipes don't have instructions so the index was out of bounds. I implemented a fix by handling this.

![image](https://user-images.githubusercontent.com/26766108/56094165-e336cb80-5ec8-11e9-8f5e-14cd100e4cf6.png)
